### PR TITLE
Merge pull request #16117 from Kojoley/ci-appveyor-unify-reqs

### PR DESCRIPTION
**Pull request by Kojoley**
- Do not update default Conda packages (saves 5 minutes on 3.6 job)
- Do not install Freetype2 from Conda (we do not use it anyway)
- Install dependencies via pip, the same way as it's done on Travis.
  However, `travis36minver.txt` cannot be used, because of some bugs
  in the pinned versions (something with pandas/pytz/dateutils)
- Test sphinext on Travis too

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by Kojoley** [comment]
Sigh. Having installed both Qt5 and Tk seems to produce a conflict in tests on Appveyor.
```
SKIPPED [11] C:\projects\matplotlib\lib\matplotlib\tests\test_backend_qt.py:32: Qt5 binding already imported
SKIPPED [1] C:\projects\matplotlib\lib\matplotlib\testing\conftest.py:68: Failed to switch to backend TkAgg (Cannot load backend 'TkAgg' which requires the 'tk' interactive framework, as 'qt5' is currently running).
```

File: .appveyor.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ test_script:
   - '"%DUMPBIN%" /DEPENDENTS lib\matplotlib\ft2font*.pyd | findstr freetype.*.dll && exit /b 1 || exit /b 0'
 
   # this are optional dependencies so that we don't skip so many tests...
-  - if x%TEST_ALL% == xyes conda install -q ffmpeg inkscape miktex pillow
+  - if x%TEST_ALL% == xyes conda install -q ffmpeg inkscape miktex
```
Comment (NelleV) [inline .appveyor.yml]: We seem to have lost pillow here as an optional dependency. Do you have a sense of many tests are not being ran as a result?

Comment (Kojoley) [reply .appveyor.yml]: We do not. It is installed via pip from `travis_all.txt`.

Comment (NelleV) [reply .appveyor.yml]: Great! Thanks for the clarification :)

File: .appveyor.yml [hunk 2]
```diff
@@ -49,22 +50,21 @@ install:
...
-  - activate
-  - conda info -a
 
   # For building, use a new environment which only includes the requirements for mpl
   # if conda-forge gets a new pyqt, it might be nice to install it as well to have more backends
   # https://github.com/conda-forge/conda-forge.github.io/issues/157#issuecomment-223536381
   - conda create -q -n test-environment python=%PYTHON_VERSION%
-    freetype=2.6 tk=8.5
-    pip setuptools numpy sphinx tornado
+    pip setuptools wheel libpng zlib tk
```
Comment (anntzer) [inline .appveyor.yml]: We (should) no longer depend on libpng and zlib anymore.

File: .appveyor.yml (hunk 1 above)
Comment (anntzer) [reply .appveyor.yml]: Actually, I think we should just delete cycler/numpy/pillow from travis_all.txt, given that we can rely on pip to install them (they're all straight dependencies).  (and travis36minver will pin the oldest versions we want to test on that single test instance)

Comment (Kojoley) [reply .appveyor.yml]: That makes sense, but I will leave it for other PR.

File: .appveyor.yml (hunk 2 above)
Comment (Kojoley) [reply .appveyor.yml]: I noticed that after I made the PR, but forgot to remove them. Done.

File: .appveyor.yml [hunk 3]
```diff
@@ -49,22 +50,21 @@ install:
...
-  - activate
-  - conda info -a
 
   # For building, use a new environment which only includes the requirements for mpl
   # if conda-forge gets a new pyqt, it might be nice to install it as well to have more backends
   # https://github.com/conda-forge/conda-forge.github.io/issues/157#issuecomment-223536381
   - conda create -q -n test-environment python=%PYTHON_VERSION%
-    freetype=2.6 tk=8.5
-    pip setuptools numpy sphinx tornado
+    pip setuptools wheel tk
```
Comment (anntzer) [inline .appveyor.yml]: also, conda's python comes with pip and setuptools by default (can be checked with `conda info pip`) -- this should be the case of most pythons anyways (unless split into pieces by a distro packagern basically).

File: .appveyor.yml [hunk 4]
```diff
@@ -49,22 +50,21 @@ install:
...
   - activate test-environment
   - echo %PYTHON_VERSION% %TARGET_ARCH%
-  - pip install -r requirements/testing/travis_all.txt -r requirements/testing/travis36.txt
+  # Install dependencies from PyPI.
+  - python -mpip install --upgrade -r requirements/testing/travis_all.txt %EXTRAREQS% %PINNEDVERS%
+  # Install optional dependencies from PyPI.
+  # Sphinx is needed to run sphinxext tests
+  - python -mpip install --upgrade sphinx
```
Comment (anntzer) [inline .appveyor.yml]: should this just go into EXTRAREQS?

File: .travis.yml [hunk 5]
```diff
@@ -133,6 +133,8 @@ install:
     python -mpip install --upgrade $PRE -r requirements/testing/travis_all.txt $EXTRAREQS $PINNEDVERS
   - |
     # Install optional dependencies from PyPI.
+    # Sphinx is needed to run sphinxext tests
+    python -mpip install --upgrade sphinx
```
Comment (anntzer) [inline .travis.yml]: as above

File: .appveyor.yml (hunk 4 above)
Comment (Kojoley) [reply .appveyor.yml]: I do not know, probably then GUIs also should go into EXTRAREQS

File: .appveyor.yml (hunk 3 above)
Comment (Kojoley) [reply .appveyor.yml]: I remember I tried to remove them and it did not work, but lets try it again

Comment (Kojoley) [reply .appveyor.yml]: Runs fine

**Comment by NelleV** [comment]
@tacaswell @anntzer Are we good on merging this PR

**Comment by QuLogic** [comment]
@anntzer I don't understand why you changed the PR title that way?

**Comment by anntzer** [comment]
Oops, sorry, this was due to https://github.com/sindresorhus/refined-github/issues/2713.  Deactivated it, and renamed.

**Pull request merged**